### PR TITLE
Enzyme: add make_zero of cuarrays

### PR DESCRIPTION
**Pull request by wsmoses**
[only downstream]

### CODE REVIEW

**Comment by maleadt** [comment]
Enzyme CI fails.

**Comment by wsmoses** [comment]
@maleadt okay fixed!

**Comment by maleadt** [comment]
Still fails. Please have a look at the CI logs.

**Comment by wsmoses** [comment]
oh apologies, I just saw the green and assumed it was fine

**Comment by maleadt** [comment]
The last build failed to even precompile the extension:

```
      From worker 2:	ERROR: LoadError: UndefVarError: `FT` not defined
      From worker 2:	Stacktrace:
      From worker 2:	 [1] top-level scope
      From worker 2:	   @ /var/lib/buildkite-agent/builds/gpuci-15/julialang/cuda-dot-jl/ext/EnzymeCoreExt.jl:583
      From worker 2:	 [2] include
      From worker 2:	   @ ./Base.jl:495 [inlined]
      From worker 2:	 [3] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt128}}, source::String)
      From worker 2:	   @ Base ./loading.jl:2285
      From worker 2:	 [4] top-level scope
      From worker 2:	   @ stdin:3
      From worker 2:	in expression starting at /var/lib/buildkite-agent/builds/gpuci-15/julialang/cuda-dot-jl/ext/EnzymeCoreExt.jl:2
      From worker 2:	in expression starting at stdin:3
```

A rebase won't fix this, so cancelling the outstanding builds.

@wsmoses Please verify locally things work first.

**Comment by wsmoses** [comment]
Fixed/verified locally

```julia
Test Summary: |  Pass  Broken  Total  Time
  Overall     | 27434      11  27445      
    SUCCESS
     Testing CUDA tests passed 

(CUDA) pkg> ^C

julia> 
```

**Comment by maleadt** [comment]
Testing now exceeds the 30min limit.

**Comment by wsmoses** [comment]
trying it with an hour to see if it's fine

**Comment by wsmoses** [comment]
however, it looks like this was already happening previously, so not due to this PR presumably: https://buildkite.com/julialang/cuda-dot-jl/builds/5616#0193de0a-806e-4e36-af34-2abfe9fa437d

**Comment by maleadt** [comment]
Testing still fails

```
Error in testset extensions/enzyme:
Error During Test at /var/lib/buildkite-agent/builds/gpuci-5/julialang/cuda-dot-jl/test/extensions/enzyme.jl:10
  Got exception outside of a @test
  UndefVarError: `T` not defined
  Stacktrace:
    [1] make_zero!
      @ /var/lib/buildkite-agent/builds/gpuci-5/julialang/cuda-dot-jl/ext/EnzymeCoreExt.jl:593 [inlined]
 ```

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -246,8 +246,7 @@ steps:
           build.message !~ /\[only/ && !build.pull_request.draft &&
             build.message !~ /\[skip tests\]/ &&
             build.message !~ /\[skip downstream\]/
-        timeout_in_minutes: 30
-        soft_fail: true
```
Comment (maleadt) [inline .buildkite/pipeline.yml:250]: This is wrong: https://github.com/JuliaGPU/CUDA.jl/commit/6bdb146a755cab9d5f5fdbe703faedabf794ed37

**Review by maleadt — CHANGES_REQUESTED** [review]

File: .buildkite/pipeline.yml (hunk 1 above)
Comment (wsmoses) [reply .buildkite/pipeline.yml:250]: Yeah okay will fix, but also I’m now confused from a high level. There an obvious undef var usage in the patch (which I will fix), which breaks CI but I never hit locally just running test.

is this not run as part of the default tests?

**Comment by wsmoses** [comment]
@maleadt okay all properly passes now (and runs in 31mins it seems)

**Pull request merged**